gwcs/wcs.py

```diff
@@ -634,3 +641,273 @@ def fix_inputs(self, fixed):
+        # from the specified pixel error. Computed at the center of the array.
+        x0, y0 = ntransform(0, 0)
+        xx, xy = ntransform(1, 0)
+        yx, yy = ntransform(0, 1)
+        dist1 = np.sqrt((xx - x0)**2 + (xy - y0)**2)
+        dist2 = np.sqrt((yx - x0)**2 + (yy - y0)**2)
+        # Average the distances in the two different directions.
+        plate_scale = (dist1 + dist2) / 2.
+        max_error = max_pix_error * plate_scale
+        # The fitting section.
+        fit_poly_x, fit_poly_y, max_resid = _fit_2D_poly(ntransform, npoints,
+                                                             degree, max_error,
+                                                             u, v, undist_x, undist_y,
+                                                             ud, vd, undist_xd, undist_yd,
+                                                             verbose=verbose)
+        # The Following is necessary to put the fit into the SIP formalism.
+        cdmat, sip_poly_x, sip_poly_y = _reform_poly_coefficients(fit_poly_x, fit_poly_y)
+        # cdmat = np.array([[fit_poly_x.c1_0.value, fit_poly_x.c0_1.value],
+        #                   [fit_poly_y.c1_0.value, fit_poly_y.c0_1.value]])
+        det = cdmat[0][0] * cdmat[1][1] - cdmat[0][1] * cdmat[1][0]
+        U = ( cdmat[1][1] * undist_x - cdmat[0][1] * undist_y) / det
+        V = (-cdmat[1][0] * undist_x + cdmat[0][0] * undist_y) / det
+        detd = cdmat[0][0] * cdmat[1][1] - cdmat[0][1] * cdmat[1][0]
+        Ud = ( cdmat[1][1] * undist_xd - cdmat[0][1] * undist_yd) / detd
+        Vd = (-cdmat[1][0] * undist_xd + cdmat[0][0] * undist_yd) / detd
+  
+        if max_inv_pix_error:
+            fit_inv_poly_u, fit_inv_poly_v, max_inv_resid = _fit_2D_poly(ntransform,
+                                                            npoints, None,
+                                                            max_inv_pix_error,
+                                                            U, V, u-U, v-V,
+                                                            Ud, Vd, ud-Ud, vd-Vd,
+                                                            verbose=verbose)
+        pdegree = fit_poly_x.degree
+        if pdegree > 1:
+            hdr['a_order'] = pdegree
+            hdr['b_order'] = pdegree
+            _store_2D_coefficients(hdr, sip_poly_x, 'A')
+            _store_2D_coefficients(hdr, sip_poly_y, 'B')
+            if max_inv_pix_error:
+                _store_2D_coefficients(hdr, fit_inv_poly_u, 'AP', keeplinear=True)
+                _store_2D_coefficients(hdr, fit_inv_poly_v, 'BP', keeplinear=True)
+            hdr['sipmxerr'] = (max_resid * plate_scale, 'Max diff from GWCS (equiv pix).')
+            if max_inv_pix_error:
+                ipdegree = fit_inv_poly_u.degree
+                hdr['ap_order'] = ipdegree
+                hdr['bp_order'] = ipdegree
+                hdr['sipiverr'] = (max_inv_resid, 'Max diff for inverse (pixels)')
```

Comment:
```suggestion
            hdr['sipmxerr'] = (max_resid * plate_scale, 'Max diff from GWCS (equiv pix).')
            if max_inv_pix_error:
                _store_2D_coefficients(hdr, fit_inv_poly_u, 'AP', keeplinear=True)
                _store_2D_coefficients(hdr, fit_inv_poly_v, 'BP', keeplinear=True)
                ipdegree = fit_inv_poly_u.degree
                hdr['ap_order'] = ipdegree
                hdr['bp_order'] = ipdegree
                hdr['sipiverr'] = (max_inv_resid, 'Max diff for inverse (pixels)')
```

Reply:
Done, sort of. I kept the same consistency for where the error appeared in the header as for the inverse.